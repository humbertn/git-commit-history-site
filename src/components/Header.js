import { Navbar, Container} from 'react-bootstrap';

function Header() {
  return (
    <>
      <Navbar bg="primary" data-bs-theme="dark">
        <Container>
          <Navbar.Brand href="#home">Gerson Estrada</Navbar.Brand>          
        </Container>
      </Navbar>
    </>
  );
}

export default Header;