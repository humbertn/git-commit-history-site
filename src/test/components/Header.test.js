import React from 'react';
import { render } from '@testing-library/react';
import Header from '../../components/Header';

describe('Header Component', () => {
    it('renders the Navbar with the correct brand text', () => {
      const { getByText } = render(<Header />);
      const brandElement = getByText('Gerson Estrada');
      expect(brandElement).toBeInTheDocument();
    });
  
    it('renders the Navbar with the correct background color', () => {
      const { container } = render(<Header />);
      const navbar = container.querySelector('.navbar.bg-primary');
      expect(navbar).toBeInTheDocument();
    });
  
    it('renders the Navbar inside a Container', () => {
      const { container } = render(<Header />);
      const containerElement = container.querySelector('.container');
      expect(containerElement).toBeInTheDocument();
    });
  
    it('matches the snapshot', () => {
      const { container } = render(<Header />);
      expect(container).toMatchSnapshot();
    });
});
  