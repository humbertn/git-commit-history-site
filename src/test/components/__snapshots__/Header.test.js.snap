// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches the snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand navbar-light bg-primary"
    data-bs-theme="dark"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="#home"
      >
        Gerson Estrada
      </a>
    </div>
  </nav>
</div>
`;
